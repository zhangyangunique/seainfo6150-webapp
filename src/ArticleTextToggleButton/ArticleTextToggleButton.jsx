import React from "react";
import styles from "./ArticleTextToggleButton.module.css";

const ArticleTextToggleButton = (props) => {
	return (
		<button className={styles.button} onClick={props.onClick}>
			{props.buttonText}
		</button>
	)
};

export default ArticleTextToggleButton;