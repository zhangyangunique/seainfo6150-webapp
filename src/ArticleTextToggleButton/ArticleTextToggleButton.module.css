.button {
	background: #c00;
	border-radius: 10px;
	color: #fff;
	padding: 10px;
}