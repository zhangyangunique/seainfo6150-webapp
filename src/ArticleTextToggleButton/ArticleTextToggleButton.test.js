import React from "react";
import ArticleTextToggleButton from "./ArticleTextToggleButton";
import { fireEvent } from "@testing-library/react";

describe("ArticleTextToggleButton texts", () => {
	it("renders correctly", () => {
		const { container } = render(<ArticleTextToggleButton buttonText={"Show more"} />);
		expect(container).toMatchSnapshot();
	});

	it("renders correctly after button is clicked", () => {
		const { container, getByText } = render(<ArticleTextToggleButton buttonText={"Show more"} />);

		const showMoreButton = getByText("Show more");
		fireEvent.click(showMoreButton);
		expect(container).toMatchSnapshot();

		const showLessButton = getByText("Show less");
		fireEvent.click(showLessButton);
		expect(container).toMatchSnapshot();
	});
}); 