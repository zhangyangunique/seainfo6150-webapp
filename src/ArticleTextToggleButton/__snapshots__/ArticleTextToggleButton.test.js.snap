// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleTextToggleButton texts renders correctly 1`] = `
<div>
  <button
    class="button"
  >
    Show less
  </button>
</div>
`;

exports[`ArticleTextToggleButton texts renders correctly after button is clicked 1`] = `
<div>
  <button
    class="button"
  >
    Show more
  </button>
</div>
`;
